### ISSUE TITLE
interpolateObject doesn’t coerce the starting value to an object.

### ISSUE THREAD

**Issue by Sumbera** [mbostock/d3#2688]
in case there is a =NaN and b is object, d3.interpolators will anyway evaluate it as d3_interpolateObject  and will subsequently fail there in if (!(k in a)) ... "Invalid Operand to 'in' : Object expected.

proposed fix that works for me was to add test for 'a' as object too on this line:
https://github.com/mbostock/d3/blob/master/src/interpolate/interpolate.js#L23
```javascript
d3.interpolators = [ function(a, b) {
      var t = typeof b;
      return  (t === "string" ? (d3_rgb_names.has(b.toLowerCase()) || /^(#|rgb\(|hsl\()/i.test(b) ? d3_interpolateRgb : d3_interpolateString)
               : b instanceof d3_color ? d3_interpolateRgb
               : Array.isArray(b) ? d3_interpolateArray
               : t === "object" && isNaN(b)  && typeof a === "object" && isNaN(a) ? d3_interpolateObject
               : d3_interpolateNumber)(a, b);
  } ];
```
v d3-3.5.12.

**Comment by mbostock** [comment]
The interpolator is supposed to be chosen entirely based on the target value, b. So the problem is that interpolateObject isn’t coercing the starting value to an object. I’ll file a bug in d3-interpolate.

**Comment by mbostock** [comment]
This was fixed in [d3-interpolate 0.4](https://github.com/d3/d3-interpolate/releases/tag/v0.4.0).
